src/tools/routed.cpp

```diff
@@ -12,6 +12,7 @@
 #include <boost/algorithm/string/case_conv.hpp>
 #include <boost/any.hpp>
 #include <boost/filesystem.hpp>
+#include <boost/optional/optional_io.hpp>
```

Comment:
Why do we need `optional_io.hpp` here? I think you want the `optional.hpp` header from boost; `optional_io.hpp` is just for printing an optional to stdout.